ee/src/main/java/org/jboss/as/ee/subsystem/ManagedExecutorServiceAdd.java

```diff
@@ -105,25 +103,23 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         final AbstractManagedExecutorService.RejectPolicy rejectPolicy = AbstractManagedExecutorService.RejectPolicy.valueOf(ManagedExecutorServiceResourceDefinition.REJECT_POLICY_AD.resolveModelAttribute(context, model).asString());
 
         final ManagedExecutorServiceService service = new ManagedExecutorServiceService(name, jndiName, hungTaskThreshold, longRunningTasks, coreThreads, maxThreads, keepAliveTime, keepAliveTimeUnit, threadLifeTime, queueLength, rejectPolicy);
-        final ServiceBuilder<ManagedExecutorServiceAdapter> serviceBuilder = context.getServiceTarget().addService(ConcurrentServiceNames.getManagedExecutorServiceServiceName(name), service);
-
-        final boolean rcPresent = context.hasOptionalCapability("org.wildfly.request-controller", null, null);
+        final CapabilityServiceBuilder serviceBuilder = context.getCapabilityServiceTarget().addCapability(ManagedExecutorServiceResourceDefinition.CAPABILITY, service);
         String contextService = null;
         if(model.hasDefined(ManagedExecutorServiceResourceDefinition.CONTEXT_SERVICE)) {
             contextService = ManagedExecutorServiceResourceDefinition.CONTEXT_SERVICE_AD.resolveModelAttribute(context, model).asString();
         }
         if (contextService != null) {
-            serviceBuilder.addDependency(ConcurrentServiceNames.getContextServiceServiceName(contextService), ContextServiceImpl.class, service.getContextServiceInjector());
+            serviceBuilder.addCapabilityRequirement(ContextServiceResourceDefinition.CAPABILITY.getName(), ContextServiceImpl.class, service.getContextServiceInjector(), contextService);
```

Comment:
This breaks the contract as the `ContextServiceResourceDefinition.CAPABILITY` is set for a type of `org.jboss.as.ee.concurrent.ContextServiceImpl` yet the type being used here is `org.glassfish.enterprise.concurrent.ContextServiceImpl`.

Reply:
nice catch

Reply:
fixed

Reply:
I would instead suggest using org.glassfish.enterprise.concurrent.ContextServiceImpl as the capability service type if possible - that way any public consumers don't need to import the subsystem module, just the ri module.

Reply:
if it ever becomes public then it probably should not use the RI at all, and that's why I have prefered to stick with the "old" service value types